src/grt/src/GlobalRouter.cpp

```diff
@@ -620,13 +622,84 @@ void GlobalRouter::updateDirtyNets(std::vector<Net*>& dirty_nets)
         logger_->error(
             GRT, 267, "Net {} has disconnected segments.", net->getName());
       }
+    } else if (net->getPins().size() >= 2) {
+      shrinkNetRoute(db_net);
     }
     net->setMergedNet(false);
     net->setDirtyNet(false);
   }
   dirty_nets_.clear();
 }
 
+void GlobalRouter::shrinkNetRoute(odb::dbNet* db_net)
+{
+  Net* net = db_net_map_[db_net];
+  GRoute segments = routes_[db_net];
+  int total_segments = segments.size();
+  std::vector<Pin> pins = net->getPins();
+
+  std::string dump;
+  netIsCovered(db_net, dump);
+  int root = -1;
+  std::vector<bool> coversPin(total_segments, false);
+
+  for (int s = 0; s < total_segments; s++) {
+    for (Pin& pin : pins) {
+      if (segmentCoversPin(segments[s], pin)) {
+        coversPin[s] = true;
+        if (pin.isDriver()) {
+          root = s;
+        }
+      }
+    }
+  }
+
+  std::vector<std::vector<int>> graph = buildNetGraph(db_net);
+
+  // Runs a BFS trough the graph
+  std::vector<uint16_t> parent(total_segments, UINT16_MAX),
+      total_children(total_segments, 0);
```

Comment:
Separate the vector declarations into two individual declarations. I find it a bit confusing the way it is right now.